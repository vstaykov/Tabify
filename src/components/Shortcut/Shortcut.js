import React from "react";
import PropTypes from "prop-types";

const Shortcut = ({ title, keys }) => (
  <div>
    <div className="shortcut-title d-inline">{title}:</div>
    {keys.map((key, i) => {
      const keyElement = (
        <span className="shortcut-key text-monospace">{key}</span>
      );
      let resultElement = (
        <span key={key}>
          {keyElement}
          <span className="shortcut-key-concatinator">+</span>
        </span>
      );

      if (i === keys.length - 1) {
        resultElement = keyElement;
      }

      return resultElement;
    })}
  </div>
);

Shortcut.propTypes = {
  title: PropTypes.string.isRequired,
  keys: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default Shortcut;
