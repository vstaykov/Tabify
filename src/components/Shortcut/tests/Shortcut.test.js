import React from "react";
import { shallow } from "enzyme";
import Shortcut from "../Shortcut";

describe("<Shortcut />", () => {
  describe("render()", () => {
    describe("when given one key", () => {
      it("render key correctly", () => {
        const props = {
          title: "Foo",
          keys: ["Ctrl"]
        };
        const component = shallow(<Shortcut {...props} />);

        expect(component).toMatchSnapshot();
      });
    });

    describe("when given several keys", () => {
      it("render all keys correctly", () => {
        const props = {
          title: "Foo",
          keys: ["Ctrl", "Shift", "Alt", "Q"]
        };
        const component = shallow(<Shortcut {...props} />);

        expect(component).toMatchSnapshot();
      });
    });
  });
});
