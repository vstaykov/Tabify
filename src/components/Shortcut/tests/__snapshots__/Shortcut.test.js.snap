// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shortcut /> render() when given one key render key correctly 1`] = `
<div>
  <div
    className="shortcut-title d-inline"
  >
    Foo
    :
  </div>
  <span
    className="shortcut-key text-monospace"
  >
    Ctrl
  </span>
</div>
`;

exports[`<Shortcut /> render() when given several keys render all keys correctly 1`] = `
<div>
  <div
    className="shortcut-title d-inline"
  >
    Foo
    :
  </div>
  <span
    key="Ctrl"
  >
    <span
      className="shortcut-key text-monospace"
    >
      Ctrl
    </span>
    <span
      className="shortcut-key-concatinator"
    >
      +
    </span>
  </span>
  <span
    key="Shift"
  >
    <span
      className="shortcut-key text-monospace"
    >
      Shift
    </span>
    <span
      className="shortcut-key-concatinator"
    >
      +
    </span>
  </span>
  <span
    key="Alt"
  >
    <span
      className="shortcut-key text-monospace"
    >
      Alt
    </span>
    <span
      className="shortcut-key-concatinator"
    >
      +
    </span>
  </span>
  <span
    className="shortcut-key text-monospace"
  >
    Q
  </span>
</div>
`;
